fe/fe-core/src/main/java/com/starrocks/external/hive/events/MetastoreEvent.java

```diff
@@ -0,0 +1,193 @@
+ */
+public abstract class MetastoreEvent {
+    private static final Logger LOG = LogManager.getLogger(MetastoreEvent.class);
+    // String.format compatible string to prepend event id and type
+    private static final String STR_FORMAT_EVENT_ID_TYPE = "EventId: %d EventType: %s ";
+
+    // logger format compatible string to prepend to a log formatted message
+    private static final String LOG_FORMAT_EVENT_ID_TYPE = "EventId: {} EventType: {} ";
+
+    // the notification received from metastore which is processed by this
+    protected final NotificationEvent event;
+
+    // dbName from the event
+    protected final String dbName;
+
+    // tblName from the event
+    protected final String tblName;
+
+    // eventId of the event. Used instead of calling getter on event everytime
+    private final long eventId;
+
+    // eventType from the NotificationEvent
+    private final MetastoreEventType eventType;
+
+    // Actual notificationEvent object received from Metastore
+    protected final NotificationEvent metastoreNotificationEvent;
+
+    // The cached instance of this event that needs to be updated
+    protected final HiveMetaCache cache;
+
+    protected MetastoreEvent(NotificationEvent event, HiveMetaCache metaCache) {
+        this.event = event;
+        this.dbName = event.getDbName();
+        this.tblName = event.getTableName();
+        this.eventId = event.getEventId();
+        this.eventType = MetastoreEventType.from(event.getEventType());
+        this.metastoreNotificationEvent = event;
+        this.cache = metaCache;
+    }
+
+    public long getEventId() {
+        return eventId;
+    }
+
+    public MetastoreEventType getEventType() {
+        return eventType;
+    }
+
+    public String getDbName() {
+        return dbName;
+    }
+
+    public String getTblName() {
+        return tblName;
+    }
+
+    /**
+     * Checks if the given event can be batched into this event. Default behavior is
+     * to return false which can be overridden by a sub-class.
+     * The current version is relatively simple to process batch events, so all that need to be processed are true.
+     *
+     * @param event The event under consideration to be batched into this event.
+     * @return false if event cannot be batched into this event; otherwise true.
+     */
+    protected boolean canBeBatched(MetastoreEvent event) {
+        return false;
+    }
+
+    /**
+     * Adds the given event into the batch of events represented by this event. Default
+     * implementation is to return null. Sub-classes must override this method to
+     * implement batching.
+     *
+     * @param event The event which needs to be added to the batch.
+     * @return The batch event which represents all the events batched into this event
+     * until now including the given event.
+     */
+    protected MetastoreEvent addToBatchEvents(MetastoreEvent event) {
+        return null;
+    }
```

Comment:
the canBeBatch has filtered the events. If it is not supported, it will not go here. Of course, the canBeBatch interface and addToBatchEvents will be modified after the schema is supported later.